roachtest: acceptance/multitenant failed

roachtest.acceptance/multitenant [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=5202348&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=5202348&tab=artifacts#/acceptance/multitenant) on master @ [ce66acdbba801de88f1dd645eaedeb3834f23dbd](https://github.com/cockroachdb/cockroach/commits/ce66acdbba801de88f1dd645eaedeb3834f23dbd):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /artifacts/acceptance/multitenant/run_1
	sql_runner.go:148,sql_runner.go:177,multitenant_upgrade.go:314,multitenant.go:46,acceptance.go:107,test_runner.go:876: error executing 'SELECT * FROM foo LIMIT 1': pq: failed to resolve n4: unable to look up descriptor for nsql4
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*acceptance/multitenant.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-15156

roachtest.acceptance/multitenant [failed](https://teamcity.cockroachdb.com/buildConfiguration/BUILDTYPE_ID-not-found-in-env/5213353?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/BUILDTYPE_ID-not-found-in-env/5213353?buildTab=artifacts#/acceptance/multitenant) on master @ [92947c29c55ff909f50a5e625811d34a1bbe71f7](https://github.com/cockroachdb/cockroach/commits/92947c29c55ff909f50a5e625811d34a1bbe71f7):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /artifacts/acceptance/multitenant/run_1
	sql_runner.go:148,sql_runner.go:177,multitenant_upgrade.go:314,multitenant.go:46,acceptance.go:107,test_runner.go:876: error executing 'SELECT * FROM foo LIMIT 1': pq: failed to resolve n4: unable to look up descriptor for nsql4
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*acceptance/multitenant.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.acceptance/multitenant [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/5223713?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/5223713?buildTab=artifacts#/acceptance/multitenant) on master @ [7dd1c51f6b5802e32bafd82e46747f349836592f](https://github.com/cockroachdb/cockroach/commits/7dd1c51f6b5802e32bafd82e46747f349836592f):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /artifacts/acceptance/multitenant/run_1
	sql_runner.go:148,sql_runner.go:177,multitenant_upgrade.go:314,multitenant.go:46,acceptance.go:108,test_runner.go:884: error executing 'SELECT * FROM foo LIMIT 1': pq: failed to resolve n4: unable to look up descriptor for nsql4
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*acceptance/multitenant.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.acceptance/multitenant [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/5230929?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/5230929?buildTab=artifacts#/acceptance/multitenant) on master @ [ba1b0d2cd8557620505d5a330bb682ce82be6c2a](https://github.com/cockroachdb/cockroach/commits/ba1b0d2cd8557620505d5a330bb682ce82be6c2a):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /artifacts/acceptance/multitenant/run_1
	sql_runner.go:148,sql_runner.go:177,multitenant_upgrade.go:314,multitenant.go:46,acceptance.go:108,test_runner.go:884: error executing 'SELECT * FROM foo LIMIT 1': pq: failed to resolve n4: unable to look up descriptor for nsql4
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*acceptance/multitenant.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.acceptance/multitenant [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/5238643?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/5238643?buildTab=artifacts#/acceptance/multitenant) on master @ [5c400e31f33b97a7b3232bb54abdd55645f0e978](https://github.com/cockroachdb/cockroach/commits/5c400e31f33b97a7b3232bb54abdd55645f0e978):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /artifacts/acceptance/multitenant/run_1
	sql_runner.go:148,sql_runner.go:177,multitenant_upgrade.go:314,multitenant.go:46,acceptance.go:108,test_runner.go:884: error executing 'SELECT * FROM foo LIMIT 1': pq: failed to resolve n4: unable to look up descriptor for nsql4
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*acceptance/multitenant.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.acceptance/multitenant [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/5250343?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/5250343?buildTab=artifacts#/acceptance/multitenant) on master @ [48e48db89eb3cd6bc38f3631364c516181811280](https://github.com/cockroachdb/cockroach/commits/48e48db89eb3cd6bc38f3631364c516181811280):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /artifacts/acceptance/multitenant/run_1
	sql_runner.go:148,sql_runner.go:177,multitenant_upgrade.go:314,multitenant.go:46,acceptance.go:108,test_runner.go:884: error executing 'SELECT * FROM foo LIMIT 1': pq: failed to resolve n4: unable to look up descriptor for nsql4
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*acceptance/multitenant.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.acceptance/multitenant [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/5264577?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/5264577?buildTab=artifacts#/acceptance/multitenant) on master @ [12198a51408e7333cd4f96b221e6734239479765](https://github.com/cockroachdb/cockroach/commits/12198a51408e7333cd4f96b221e6734239479765):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /artifacts/acceptance/multitenant/run_1
	sql_runner.go:148,sql_runner.go:177,multitenant_upgrade.go:314,multitenant.go:46,acceptance.go:108,test_runner.go:884: error executing 'SELECT * FROM foo LIMIT 1': pq: failed to resolve n4: unable to look up descriptor for nsql4
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*acceptance/multitenant.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/cmd/roachtest/tests/acceptance.go
+++ b/pkg/cmd/roachtest/tests/acceptance.go
@@ -53,7 +53,6 @@ func registerAcceptance(r registry.Registry) {
 		registry.OwnerMultiTenant: {
 			{
 				name: "multitenant",
-				skip: "https://github.com/cockroachdb/cockroach/issues/81506",
 				fn:   runAcceptanceMultitenant,
 			},
 		},

--- a/pkg/cmd/roachtest/tests/multitenant_upgrade.go
+++ b/pkg/cmd/roachtest/tests/multitenant_upgrade.go
@@ -42,10 +42,12 @@ func registerMultiTenantUpgrade(r registry.Registry) {
 // Sketch of the test:
 //
 //  * Host{Binary: Prev, Cluster: Prev}: Start host cluster.
-//  * Tenant11{Binary: Prev, Cluster: Prev}: Create tenant 11 and verify it works.
+//  * Host{Binary: Prev, Cluster: Prev}: Run create_tenant(11) and create_tenant(12).
+//  * Tenant11{Binary: Prev, Cluster: Prev}: Start tenant 11 and verify it works.
 //  * Host{Binary: Cur, Cluster: Prev}: Upgrade host cluster (don't finalize).
 //  * Tenant11{Binary: Prev, Cluster: Prev}: Verify tenant 11 still works.
-//  * Tenant12{Binary: Prev, Cluster: Prev}: Create tenant 12 and verify it works.
+//  * Tenant12{Binary: Prev, Cluster: Prev}: Start tenant 12 and verify it works.
+//  * Host{Binary: Cur, Cluster: Prev}: Run create_tenant(13).
 //  * Tenant13{Binary: Cur, Cluster: Prev}: Create tenant 13 and verify it works.
 //  * Tenant11{Binary: Cur, Cluster: Prev}: Upgrade tenant 11 binary and verify it works.
 //  * Tenant11{Binary: Cur, Cluster: Cur}: Run the version upgrade for the tenant 11.
@@ -102,7 +104,7 @@ func runMultiTenantUpgrade(ctx context.Context, t test.Test, c cluster.Cluster, 
 			withResults([][]string{{initialVersion}}),
 	)
 
-	t.Status("preserving downgrade option on host server")
+	t.Status("preserving downgrade option on system tenant")
 	{
 		s := runner.QueryStr(t, `SHOW CLUSTER SETTING version`)
 		runner.Exec(
@@ -111,23 +113,22 @@ func runMultiTenantUpgrade(ctx context.Context, t test.Test, c cluster.Cluster, 
 		)
 	}
 
-	t.Status("upgrading host server")
+	t.Status("creating a new tenant 12")
+	const tenant12HTTPPort, tenant12SQLPort = 8012, 20012
+	const tenant12ID = 12
+	runner.Exec(t, `SELECT crdb_internal.create_tenant($1)`, tenant12ID)
+
+	t.Status("upgrading system tenant binary")
 	c.Stop(ctx, t.L(), option.DefaultStopOpts(), kvNodes)
 	settings.Binary = currentBinary
 	c.Start(ctx, t.L(), option.DefaultStartOpts(), settings, kvNodes)
 	time.Sleep(time.Second)
 
-	t.Status("checking the pre-upgrade sql server still works after the KV binary upgrade")
+	t.Status("checking the pre-upgrade sql server still works after the system tenant binary upgrade")
 
 	verifySQL(t, tenant11.pgURL,
 		mkStmt(`SELECT * FROM foo LIMIT 1`).
 			withResults([][]string{{"1", "bar"}}))
-
-	t.Status("creating a new tenant 12")
-
-	const tenant12HTTPPort, tenant12SQLPort = 8012, 20012
-	const tenant12ID = 12
-	runner.Exec(t, `SELECT crdb_internal.create_tenant($1)`, tenant12ID)
 
 	t.Status("starting tenant 12 server with older binary")
 	tenant12 := createTenantNode(ctx, t, c, kvNodes, tenant12ID, tenantNode, tenant12HTTPPort, tenant12SQLPort, tenantStartOpt)
@@ -182,9 +183,13 @@ func runMultiTenantUpgrade(ctx context.Context, t test.Test, c cluster.Cluster, 
 				withResults([][]string{{initialVersion}}))
 	}
 
-	// Note that this is exercising a path we likely want to eliminate in the
-	// future where the tenant is upgraded before the KV nodes.
-	t.Status("migrating the tenant 11 to the current version before kv is finalized")
+	t.Status("finalizing the system tenant upgrade")
+	runner.Exec(t, `SET CLUSTER SETTING cluster.preserve_downgrade_option = DEFAULT`)
+	runner.CheckQueryResultsRetry(t,
+		"SELECT version = crdb_internal.node_executable_version() FROM [SHOW CLUSTER SETTING version]",
+		[][]string{{"true"}})
+
+	t.Status("migrating the tenant 11 to the current version after system tenant is finalized")
 
 	verifySQL(t, tenant11.pgURL,
 		mkStmt(`SELECT * FROM foo LIMIT 1`).
@@ -195,12 +200,6 @@ func runMultiTenantUpgrade(ctx context.Context, t test.Test, c cluster.Cluster, 
 		mkStmt("SELECT version = crdb_internal.node_executable_version() FROM [SHOW CLUSTER SETTING version]").
 			withResults([][]string{{"true"}}),
 	)
-
-	t.Status("finalizing the kv server")
-	runner.Exec(t, `SET CLUSTER SETTING cluster.preserve_downgrade_option = DEFAULT`)
-	runner.CheckQueryResultsRetry(t,
-		"SELECT version = crdb_internal.node_executable_version() FROM [SHOW CLUSTER SETTING version]",
-		[][]string{{"true"}})
 
 	t.Status("stopping the tenant 12 server ahead of upgrading")
 	tenant12.stop(ctx, t, c)

--- a/pkg/sql/sqlinstance/instancestorage/instancereader.go
+++ b/pkg/sql/sqlinstance/instancestorage/instancereader.go
@@ -162,6 +162,7 @@ func (r *Reader) maybeStartRangeFeed(ctx context.Context) *rangefeed.RangeFeed {
 		updateCacheFn,
 		rangefeed.WithInitialScan(initialScanDoneFn),
 		rangefeed.WithOnInitialScanError(initialScanErrFn),
+		rangefeed.WithRowTimestampInInitialScan(true),
 	)
 	r.setStarted()
 	if err != nil {

